Add samples cmd to agentctl

simonswine opened the pull request:
This helped investigating which series/samples are within a WAL file.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/agent?pullRequest=239) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/agent?pullRequest=239) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/agent?pullRequest=239) it.</sub>

rfratto commented on pkg/agentctl/samples.go:
```diff
@@ -0,0 +1,131 @@
+	Count  int64
+}
+
+// FindSamples TODO
```
Should this comment be removed? :)

simonswine replied:
It should :slightly_smiling_face: 

// TODO remove me

rfratto commented on pkg/agentctl/samples.go:
```diff
@@ -0,0 +1,131 @@
+
+	var series []*Series
+	for ref, labels := range labelsByRef {
+		series = append(
+			series,
+			&Series{
+				Labels: labels,
+				Count:  sampleCountByRef[ref],
+				MinTS:  timestamp.Time(minTSByRef[ref]),
+				MaxTS:  timestamp.Time(maxTSByRef[ref]),
+			},
+		)
```
Personally I find the indentation a little weird here, can we change it?

```suggestion
		series = append(series, &Series{
			Labels: labels,
			Count:  sampleCountByRef[ref],
			MinTS:  timestamp.Time(minTSByRef[ref]),
			MaxTS:  timestamp.Time(maxTSByRef[ref]),
		})
```

simonswine replied:
Yes that's better :slightly_smiling_face:

rfratto reviewed (commented):
Nice, I love this! One small nit but LGTM!

rfratto commented:
Oh, one more thing I always forget to mention in the review: please add this to the CHANGELOG and give yourself credit :)

rfratto commented on cmd/agentctl/main.go:
```diff
@@ -79,6 +81,45 @@ source-of-truth directory.`,
+
+	cmd := &cobra.Command{
+		Use:   "samples [WAL directory]",
+		Short: "Discover sample statistics for series matching a label selector within the WAL.",
```
Can you give this a `Long` documentation with an example of a value for the selector flag? It's not immediately clear to me how it could be used.

simonswine replied:
Done that, also ensured the default selector is matching all series

simonswine commented:
@rfratto thanks for your review, just addressed your comments. Double check everything it's the last thing I have done before the weekend :joy: 

I will certainly mention `agentctl` on the on-call hand-over, as it might be super useful for other debugging tasks around WALs.
